store/postgres/src/relational.rs

```diff
@@ -631,6 +713,28 @@ impl Layout {
     }
 }
 
+/// A user-defined enum
+#[derive(Clone, Debug, PartialEq)]
+pub struct EnumType {
+    /// The name of the Postgres enum we created, fully qualified with the schema
+    pub name: SqlName,
+    /// The possible values the enum can take
+    values: Arc<BTreeSet<String>>,
+}
+
+impl EnumType {
+    fn is_assignable_from(&self, source: &Self) -> Option<String> {
+        if source.values.is_subset(self.values.as_ref()) {
+            None
+        } else {
+            Some(format!(
+                "the enum type {} contains values not present in {}",
+                source.name, self.name
+            ))
+        }
+    }
+}
```

Comment:
The more I look at this, the more I think these checks should take place at the GraphQL schema level. That's intended to be the source of truth, and the Postgres schema is just how it's manifested at the storage layer.

Reply:
The problem I have with that is that it is not apparent at all what the truth is when you have a `Schema` object (with the parser's raw `Document` in it) Even if we decide it needs to go to the GraphQL layer right now, I'd prefer to do that as a follow-on PR rather than leaving this open until that's done.

Reply:
Yeah, agreed (see the other comment).